pkg/pillar/cmd/zedagent/parseEdgeview.go

```diff
@@ -0,0 +1,330 @@
+
+	log.Tracef("Started parsing edge-view config")
+	zcfgEv := config.GetEdgeview()
+	evConfig := types.EdgeviewConfig{}
+	if zcfgEv == nil {
+		if !cmp.Equal(ctx.configEdgeview, &evConfig) {
+			removeEvFiles()
+		}
+		ctx.configEdgeview = &evConfig
+		return
+	}
+	evConfig = types.EdgeviewConfig{
+		JWToken:     zcfgEv.Token,
+		DispCertPEM: zcfgEv.DispCertPem,
+	}
+
+	// device side
+	dev := types.EvDevPolicy{}
+	if zcfgEv.DevPolicy != nil {
+		dev = types.EvDevPolicy{
+			Enabled: zcfgEv.DevPolicy.AllowDev,
+		}
+	}
+	evConfig.DevPolicy = dev
+
+	// app side
+	app := types.EvAppPolicy{}
+	if zcfgEv.AppPolicy != nil {
+		app = types.EvAppPolicy{
+			Enabled: zcfgEv.AppPolicy.AllowApp,
+		}
+	}
+	evConfig.AppPolicy = app
+
+	changed := !cmp.Equal(ctx.configEdgeview, &evConfig)
+	if changed {
+		params := strings.SplitN(evConfig.JWToken, ".", 3)
+		if len(params) != 3 {
+			removeEvFiles()
+		} else {
+			// need to validate the signature for JWT
+			err := verifyJWT(params)
+			if err == nil {
+				err = addEvFiles(evConfig, params)
+			}
+			if err != nil {
+				removeEvFiles()
+			}
+		}
+	}
+	ctx.configEdgeview = &evConfig
+}
+
+func verifyJWT(params []string) error {
+	certBytes, err := ioutil.ReadFile(types.ServerSigningCertFileName)
+	if err != nil {
+		log.Errorf("can not read signing cert: %v", err)
+		return err
+	}
+
+	var ecdsaKey *ecdsa.PublicKey
+	if ecdsaKey, err = jwt.ParseECPublicKeyFromPEM(certBytes); err != nil {
+		log.Errorf("Unable to parse ECDSA public key: %v", err)
+		return err
+	}
+
+	var jalgo types.EvjwtAlgo
+	part1, err := base64.RawURLEncoding.DecodeString(params[0])
+	if err != nil {
+		log.Errorf("can not decode jwt algo: %v", err)
+		return err
+	}
+	err = json.Unmarshal(part1, &jalgo)
+	if err != nil {
+		log.Errorf("json unmarshal algo error: %v", err)
+		return err
+	}
+
+	method := jwt.GetSigningMethod(jalgo.Alg)
+	err = method.Verify(strings.Join(params[0:2], "."), params[2], ecdsaKey)
+	if err != nil {
+		log.Errorf("verify jwt failed: %v", err)
+		return err
+	}
+	log.Tracef("jwt verify ok")
+
+	return nil
+}
+
+func addEvFiles(evConfig types.EdgeviewConfig, params []string) error {
+	tokenData, err := base64.RawURLEncoding.DecodeString(params[1])
+	if err != nil {
+		log.Errorf("base64 decode jwt error: %v", err)
+		return err
+	}
+
+	var jdata types.EvjwtInfo
+	err = json.Unmarshal(tokenData, &jdata)
+	if err != nil {
+		log.Errorf("json unmarshal jwt error: %v", err)
+		return err
+	}
+
+	if _, err = os.Stat(types.EdgeviewPath); os.IsNotExist(err) {
+		if err := os.MkdirAll(types.EdgeviewPath, 0755); err != nil {
+			log.Errorf("failed to create dir: %v", err)
+			return err
+		}
+	}
+
+	// check devUUID
+	if devUUID.String() != jdata.Sub {
+		err := fmt.Errorf("jwt sub does not match devUUID: %s", jdata.Sub)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// checks done
+
+	// check the expiration time
+	nowSec := int(time.Now().Unix())
```

Comment:
This 32 bit integer will overflow in 2038. Is the Exp field in the JWT int32 or should it be uint64?